### PR TITLE
Feature/events

### LINKED ISSUE #3: [3.0 pt] Math magicians: events
- [ ] Handle events in a React app.

- [ ] Use React life cycle methods.

- [ ] Understand the mechanism of lifting state up.

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
- [x] Make sure that there are no linter errors.
- [x] Make sure that you used the correct Gitflow.
- [x] Make sure that you documented your work in a professional way.
### JavaScript requirements
- [x] Follow our list of best practices for JavaScript.
### Project requirements
- [x] In the [previous project](https://github.com/Benawi/Microverse-Reac-Math-Magicians/pull/5), I worked on the components of a React application using [Create React App](https://reactjs.org/docs/create-a-new-react-app.html), now in this project:
   - [x] A new directory called logic  is created inside the project
   - [x]  [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js) and [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js) files copied into logic/ directory
   - [x]  [big.js](https://www.npmjs.com/package/big.js) installed using NPM
   - [x]  [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js) and [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js) files I understand what they do
   - [x] [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js) is imported in [Calculator.js](https://github.com/Benawi/Microverse-Reac-Math-Magicians/blob/feature/Events/src/components/calculator.js) component
   - [x] Click event handlersimplemented  using the math logic from calculate.js and operate.js in your React component
   - [x] In the browser tested ; it render without problems, [this](https://github.com/Benawi/Microverse-Reac-Math-Magicians/assets/21217148/219b8efd-d227-4aba-b08a-1ef49d460565)
   - [x] React app configured for deployment to GitHub Pages [🌐Live Math Web App](https://benawi.github.io/Microverse-Reac-Math-Magicians/)

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,41 +1,80 @@
...
+      <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '+', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '+', setCalculatorObj); }}>+</div>
+      <div className="zero-button calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '0', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '0', setCalculatorObj); }}>0</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '.', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '.', setCalculatorObj); }}>.</div>
+      <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '=', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '=', setCalculatorObj); }}>=</div>
+    </div>
```
Comment (abdulhamiid) [inline src/components/calculator.js]: To follow our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md). I would suggest that we reduce the number of duplicated codes to follow the DRY principle (Do not Repeat Yourself) 🤓

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Benawi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️Handle events in a React app.
- ✔️ Use React life cycle methods.
- ✔️ Use correct Git Flow


## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
## Changes ♻️
- [x] Code refactored to reduce the number of duplicated codes which will address DRY principle 
- [x] The following changes done:
   - [x]  Array of button attrubuties is created [L40-L58](https://github.com/Benawi/Microverse-Reac-Math-Magicians/blob/fa96274aa1caa2fbdc8edd1c04d800515781e332/src/components/calculator.js#L40-L58)
   - [x] A function which creates a buttons is created [LL77C1-L87C3](https://github.com/Benawi/Microverse-Reac-Math-Magicians/blob/fa96274aa1caa2fbdc8edd1c04d800515781e332/src/components/calculator.js#LL77C1-L87C3)
   - [x] Buttons rendered according to their argument [L61-L76](https://github.com/Benawi/Microverse-Reac-Math-Magicians/blob/fa96274aa1caa2fbdc8edd1c04d800515781e332/src/components/calculator.js#L61-L76)

**Review by rustam-tolipov — APPROVED** [review]
Hi @Benawi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Your code looks great as requested :)

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**